devtools/new-changelog-release.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2020-2020 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+set -e
+set -u
+
+# this script edits the current release in the release notes or adds a new
+# unreleased section
+
+prog="$(basename "$0")"
+topdir="$(realpath "$(dirname "$0")/..")"
+pushd "$topdir" >/dev/null
+
+if [ $# -ne 2 ] && [ "${1:-}" != unreleased ]; then
+  echo "usage: $prog {<version> <date>|unreleased}" >&2
```

Comment:
What date format is required here? DD-MM-YYYY ? A hint would be helpful.

Reply:
The provided string is appended after the release-number in `CHANGELOG.md`, so `## [Unreleased]` will become `## [<version>] - <date>`. 
While this is meant to contain a date, you can pass any string you like.
As the script is meant to be called by `prepare-release.sh` and that script knows what to pass, I didn't put much effort in parameter sanitation here.